### ISSUE TITLE
Memory Usage 

### ISSUE THREAD

**Issue by JamieKitson** [samjudson/flickrnet-experimental#4]
I use FlickrNet in a WP background task which has very low memory cap. I'm trying to add the ability to upload large images and videos. I've replaced a couple of FlickrNet's MemoryStreams with FileStreams which helped, but unfortunately `HttpClient.PostAsync(uri, StreamContent)` seems to use a MemoryStream behind the scenes even if I create it with a FileStream. Do you have any idea if I can force a .Net POST to use a FileStream?

Thanks.

**Comment by samjudson** [comment]
I had this issue on the old code. The problem is I think I really need to the content-length otherwise Flickr complains, so I am reading the file into memory.

I made a change to only read the file into memory is the stream isn't seekable - because on a seekable stream you can get the length. I'll need to check the code to see if this change made it over - based on what you are saying it hasn't.

**Comment by samjudson** [comment]
Yes, I changed the `CreateUploadData` method to return a `StreamCollection` (my own class I think) rather than a `byte[]`:

https://github.com/samjudson/flickr-net/blob/f60c445af54d713ebd1e2ab9ad72a17779901587/FlickrNet/Flickr.cs#L498

The same should be done over here for this code.

**Comment by JamieKitson** [comment]
That's interesting. I though I'd gone back and tried v3 and had the same issue but I'll try again.

**Comment by JamieKitson** [comment]
In the end I had to go back to using `HttpWebRequest` and set `AllowWriteStreamBuffering = false`.

**Comment by samjudson** [comment]
Yeah, I can't find a way to work with the WebClient, as it doesn't allow opening up a request stream and a response stream. If you have something you want to submit feel free?

**Comment by Yortw** [comment]
FWIW I've heard you shouldn't use http client to handle large data transfers. If you're in WP/store apps the answer is supposed to be to use the background upload API. Its quite different and involves a fair amount of work though, and last time I looked FlickrNet didn't have method that supported it.

**Comment by JamieKitson** [comment]
I don't think my solution is portable as it uses a `FileStream`. 

https://github.com/JamieKitson/FlickrAutoUploader/blob/master/PhoneClassLibrary1/MyFlickr.cs#L307

Otherwise I just took `FlickrResponder.UploadData` from v3 and added `req.AllowWriteStreamBuffering = false`.

https://github.com/JamieKitson/FlickrAutoUploader/blob/master/PhoneClassLibrary1/MyFlickr.cs#L332
